test/spec/juttle-spec.js

```diff
@@ -155,10 +155,12 @@ class SpecRenderer {
         for (var erri=0 ; erri < test.errors.length ; erri++) {
             parts.push('            expect(res.errors[' + erri + ']).to.include(' + JSON.stringify(test.errors[erri]) + ');');
         }
+        parts.push('            expect(res.errors.length, "Expected test to emit ' + test.errors.length + ' errors: " + JSON.stringify(res.errors)).to.equal(' + test.errors.length + ');');
```

Comment:
I’d move this test above the tests validating individual errors. This will prevent non-sensical message comparisons in cases where the code produces less errors than expected.